libminifi/test/integration/IntegrationBase.h

```diff
@@ -109,6 +110,16 @@ void IntegrationBase::run(std::string test_file_location) {
 
   core::YamlConfiguration yaml_config(test_repo, test_repo, content_repo, stream_factory, configuration, test_file_location);
 
+  auto controller_service_provider = yaml_ptr->getControllerServiceProvider();
+  std::string state_dir_name_template = "/tmp/integrationstate.XXXXXX";
+  std::vector<char> state_dir_buf(state_dir_name_template.c_str(),
+                                  state_dir_name_template.c_str() + state_dir_name_template.size() + 1);
+  if (mkdtemp(state_dir_buf.data()) == nullptr) {
+    throw std::runtime_error("Failed to create temporary directory for state");
+  }
+  state_dir = state_dir_buf.data();
```

Comment:
duplicate code, consider extracting to a function

Reply:
No longer exists in the rebased version.